### ISSUE TITLE
torch_xla scan forces inputs to have gradients

### ISSUE THREAD

**Issue by tengyifei** [pytorch/xla#8783]
The snippet https://github.com/pytorch/xla/blob/00fac787e986a51e1d5d8b74603867cea4086477/torch_xla/experimental/scan.py#L217-L226 is probably wrong.

The most obvious example is that if one of the input is an integer, then it can't possibly have gradients.

**Comment by tengyifei** [comment]
After consulting the LLMs, it looks like PyTorch autograd will discard any gradient we return from the `backwards()` if that tensor is not supposed to require gradients (we should write a test for this). The fix here is that `requires_grad` should be `True` unless the tensor is not a floating point one.
